lib/Twig/Error.php

```diff
@@ -67,8 +69,9 @@ public function __construct($message, $lineno = -1, $name = null, Exception $pre
 
         $this->lineno = $lineno;
         $this->filename = $name;
+        $this->path = $path;
 
-        if (-1 === $lineno || null === $name) {
+        if (-1 === $lineno || null === $name || null === $path) {
```

Comment:
If you look at the full patch, you'll notice that when a no line is provided when building an exception, no path is provided.
And reciprocally when a line is provided, a path is always provided.
This means this added condition doesn't trgger any new "guess" in the code.
Yet, this should be added for consistency IMO, because having a partially guessed state makes no sense.